src/services/completions.ts

```diff
@@ -2831,6 +2849,27 @@ namespace ts.Completions {
 
         if (!contextToken) return undefined;
 
+        function isPropertyBeforeConstructorInitalized(classDecl: ClassDeclaration, _constructor: ClassElement) {
+            const indexOfConstructor = classDecl.members.indexOf(_constructor);
+            if (indexOfConstructor <= 0) {
+                return true;
+            }
+            return isInitializedProperty(classDecl.members[indexOfConstructor - 1]);
+        }
+
+        // class c { prop \n constructor| }
+        // class c { prop: number \n constructor| }
+        // class c { prop = somewhat \n constructor| }
+        // and also for `;` replacing `\n`
+        if (location.kind === SyntaxKind.ConstructorKeyword && (
+            isIdentifier(contextToken) ||
+            isTypeKeyword(contextToken.kind) ||
+            isPropertyBeforeConstructorInitalized(getAncestor(location, SyntaxKind.ClassDeclaration) as ClassDeclaration, location.parent as ClassElement) ||
+            contextToken.kind === SyntaxKind.SemicolonToken
+        )) {
+            return getAncestor(contextToken, SyntaxKind.ClassDeclaration) as ObjectTypeDeclaration;

```

Comment:
I _believe_ we will not produce a `ConstructorKeyword` outside a class (it will just be an `Identifier` if it appears), but the class node could be a `ClassExpression`, not only a `ClassDeclaration`.

Reply:
Got it.